libraries/AP_GPS/AP_GPS_SBP2.cpp

```diff
@@ -0,0 +1,482 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+//
+//  Swift Navigation SBP GPS driver for ArduPilot.
+//	Code by Niels Joubert
+//
+//  Swift Binary Protocol format: http://docs.swift-nav.com/
+//
+
+#include "AP_GPS.h"
+#include "AP_GPS_SBP2.h"
+#include <DataFlash/DataFlash.h>
+
+extern const AP_HAL::HAL& hal;
+
+#define SBP_DEBUGGING 1
+#define SBP_INFOREPORTING 1
+
+//INVARIANT: We expect SBP to give us a heartbeat in less than 2 seconds.
+//  This is more lax than the default Piksi settings,
+//  and we assume the user hasn't reconfigured their Piksi to longer heartbeat intervals
+#define SBP_TIMEOUT_HEARTBEAT  2000
+
+#if SBP_DEBUGGING
+ # define Debug(fmt, args ...)                  \
+do {                                            \
+    hal.console->printf("%s:%d: " fmt "\n",     \
+                        __FUNCTION__, __LINE__, \
+                        ## args);               \
+    hal.scheduler->delay(1);                    \
+} while(0)
+#else
+ # define Debug(fmt, args ...)
+#endif
+
+#if SBP_INFOREPORTING
+ # define Info(fmt, args ...)                  \
+do {                                            \
+    hal.console->printf("SBPv2: " fmt "\n",     \
+                        ## args);               \
+    hal.scheduler->delay(1);                    \
```

Comment:
I was under the impression this line is to flush the HAL printf buffers, else the print might not happen for a long time. Is there an alternative method to flush the buffers?